django/middleware/locale.py

```diff
@@ -37,6 +37,8 @@ def process_response(self, request, response):
 
         if response.status_code == 404 and not language_from_path and i18n_patterns_used:
             language_path = '/%s%s' % (language, request.path_info)
+            if i18n_patterns_used and not prefixed_default_language and str(language) == str(settings.LANGUAGE_CODE):
```

Comment:
There is no need to check `i18n_patterns_used` here, it's already checked above.
I believe the cast to `str()` is unnecessary.